docs(messaging): add token management info to server integration doc (#5427)
invertase/react-native-firebase: docs/messaging/server-integration.md

Your application code can then handle messages as you see fit; updating local cache, displaying a [notification](/messaging/notifications)
or updating UI. The possibilities are endless!

## Signing out users 

> Firebase Cloud Messaging tokens are associated with the instance of the installed app. So reinstalling the app and logging back in with the same user will generate a fresh token. 

Reviewer: Logging in with the same user will not generate a fresh token, that would be a contradiction of the first (correct) sentence here where you explain that it is tied to the installation, not the user

```suggestion
Firebase Cloud Messaging tokens are associated with the instance of the installed app. By default, only token expiration or uninstalling/reinstalling the app will generate a fresh token.
```

> Similarly, when one user logs out and another logs in on the same app on the same device, the same FCM token will be used for both users. So if the use case involves users logging in and out on the app, the tokens should be cycled in concert. 

Reviewer: ```suggestion
This means that by default, if your app has users and you allow them to log out and log in on the same app on the same device, the same FCM token will be used for both users. Usually this is not what you want, so you must take care to cycle the FCM token at the same time you handle user logout/login.
```

> To log a user out, these two methods should be called back-to-back, waiting for promise resolution of course
> 
> https://rnfirebase.io/reference/messaging#deleteToken
> 
> https://rnfirebase.io/reference/messaging#getToken
> 
> Do note that when a token is deleted by calling the `deleteToken` method, it is gone forever. 

Reviewer: ```suggestion
How and when you invalidate a token and generate a new one will be specific to your project, but a common pattern is to delete the FCM token during logout and update your back end to remove ti, then to fetch the FCM token during login and update your back end systems to associate the new token with the logged in user.

https://rnfirebase.io/reference/messaging#deleteToken
https://rnfirebase.io/reference/messaging#getToken

Note that when a token is deleted by calling the `deleteToken` method, it is immediately and permanently invalid. 
```
